docs/architecture.md

```diff
@@ -0,0 +1,31 @@
+## 🏗️ Architecture Overview
+
+Propeller's architecture consists of three key components:
+
+### 1. **User Interface (CLI/API)**
+
+- Provides users with tools to interact with Propeller, whether via a command-line interface (CLI) or RESTful API.
```

Comment:
No need for UI, let's keep this simple

Reply:
Actually, coming back to UI - let's keep it. It will be important to see network topology, see node types and have visualisation of cluster. 

UI will talk to Manager's HTTP API.